xbmc/FileItem.cpp

```diff
@@ -883,12 +883,15 @@ bool CFileItem::Exists(bool bUseCache /* = true */) const
   if (URIUtils::IsStack(strPath))
     strPath = CStackDirectory::GetFirstStackedFile(strPath);
 
-  if (m_bIsFolder)
+  if (URIUtils::IsBluray(strPath) && URIUtils::GetFileName(strPath) == "menu")
+  {
+    strPath = CURL(strPath).GetHostName();
     return CDirectory::Exists(strPath, bUseCache);
-  else
-    return CFile::Exists(strPath, bUseCache);
+  }
```

Comment:
I think the fix belongs to the bluray:// protocol handler, which should recognize the virtual "menu" file.